### ISSUE TITLE
Fps drops after inactivity

### ISSUE THREAD

**Issue by PoSayDone** [hyprwm/Hyprland#2484]
I have a 90Hz display and I get a drop in fps after a few seconds of inactivity, the fps drops to almost 60. To get normal fps again, I need to switch the workspace or make the window full screen, if I stop moving the mouse or scrolling, the fps drops again. I tried disabling the vfr option in the configuration, but to no avail.

Steps to reproduce:
- Start hyprland
- Open any app, for example firefox
- Be inactive for a second
- Start scrolling and get fps drop

Hyprland version: hyprland-0.26.0-1
Hyprland config: https://pastebin.com/raw/ACfPMGBa 
GPU: Intel Iris Xe

If any other information needs to be provided, I would be happy to do so.

**Comment by Zeioth** [comment]
Maybe you have 

    vfr = true

That's, [what it does](https://wiki.hyprland.org/Configuring/Variables/#vfr).

**Comment by PoSayDone** [comment]
> Maybe you have
> 
> ```
> vfr = true
> ```
> 
> That's, [what it does](https://wiki.hyprland.org/Configuring/Variables/#vfr).

As I said before, turning VFR off and on does not work.

I realized that the problem is that the frequency of the video core at such moments of fps drops does not rise, when the drop begin, the frequency is 300mhz, but after switching to another desktop or switching to full screen mode, it rises. So it's my power settings, although it's strange that there are no such problems in gnome.

**Comment by PoSayDone** [comment]
So, I tried to disable TLP, tried to use performance mode with the power profile daemon, but no success, so I think the problem is not in power managment. I don't have lag in GNOME or Sway, but I do with the default/non-default configuration in hyprland.

**Comment by vaxerski** [comment]
maybe its just your gpu not keeping up?

**Comment by PoSayDone** [comment]
as temporary fix I just raised min clock for gpu to 700mhz and don't have such issue now, but I still don't know why frequency is not raising properly only in hyprland

**Comment by PoSayDone** [comment]
As you can see, gpu clock is not rising properly and it rises only when I switch workspace. If I stop scrolling it would go back to low values. (gpu clock speed is higher cause of recording, usually it's 300MHz when lagging and 700MHz when everything is okay)

https://github.com/hyprwm/Hyprland/assets/29358657/d3456216-0224-4c20-9325-0960d4beab52

**Comment by vaxerski** [comment]
hyprland does not control your clocks, this looks like an issue on your side

**Comment by romanstingler** [comment]
@PoSayDone can you test with the latest git commit ?

**Comment by PoSayDone** [comment]
> @PoSayDone can you test with the latest git commit ?

sure, 10 mins

**Comment by PoSayDone** [comment]
> @PoSayDone can you test with the latest git commit ? PS: please don't enable the hyprland overlay

nah, nothing changed for me, still get low fps after some inactivity

**Comment by xcpky** [comment]
I've had the same issue since long ago, and I just realize workspace switching frame dropping is caused by the gpu schedule. see #2305 . I think they are the same issue.

**Comment by SychoAN** [comment]
for me with amd gpu 
this was the fix for this issue...you guys should try it

https://github.com/hyprwm/Hyprland/issues/3018#issuecomment-1717932050

**Comment by fulalas** [comment]
I can confirm this issue on my machine as well. It doesn't happen on any other desktop environment, either X11 or Wayland.

I have a NVidia GTX 1650 Super + 120 Hz monitor, and `vfr` is set to `false`. I also tried  `vrr = 0` but it didn't change anything.

I believe this issue is more (only?) noticeable with high refresh rate monitors.

Hyprland version `0.33.0`.

**Comment by PoSayDone** [comment]
Well, I found what causes this problem for me specifically. If you disable fractional scaling, everything starts to work smoothly and there is no drop in fps, but as soon as you enable it, everything starts to lag as before.

**Comment by izmyname** [comment]
Bump. Was it fixed on the latest git?

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!

**Comment by zyncc** [comment]
> as temporary fix I just raised min clock for gpu to 700mhz and don't have such issue now, but I still don't know why frequency is not raising properly only in hyprland

hey can you tell me how i can do the same as well?
